gym_wrapper/base_gym_env.py

```diff
@@ -47,37 +54,80 @@ def __init__(
         self.game.set_window_visible(False)
 
         screen_format = self.game.get_screen_format()
-        if screen_format != vzd.ScreenFormat.RGB24:
-            warnings.warn(f"Detected screen format {screen_format.name}. Only RGB24 is supported in the Gym wrapper. Forcing RGB24.")
+        if screen_format != vzd.ScreenFormat.RGB24 and screen_format != vzd.ScreenFormat.GRAY8:
+            warnings.warn(f"Detected screen format {screen_format.name}. Only RGB24 and GRAY8 is supported in the Gym wrapper. Forcing RGB24.")
             self.game.set_screen_format(vzd.ScreenFormat.RGB24)
 
-        self.game.init()
         self.state = None
         self.window_surface = None
         self.isopen = True
+        self.channels = 3
+        if screen_format == vzd.ScreenFormat.GRAY8:
+            self.channels = 1
 
         self.depth = self.game.is_depth_buffer_enabled()
         self.labels = self.game.is_labels_buffer_enabled()
         self.automap = self.game.is_automap_buffer_enabled()
 
-        allowed_buttons = []
+        delta_buttons = []
+        binary_buttons = []
+        button_space = {}
         for button in self.game.get_available_buttons():
-            if "DELTA" in button.name:
-                warnings.warn(f"Removing button {button.name}. DELTA buttons are currently not supported in Gym wrapper. Use binary buttons instead.")
+            if vzd.is_delta_button(button) and button not in delta_buttons:
+                max_value = self.game.get_button_max_value(button)
+                # if max value not set, default to infinity
+                if max_value == 0:
+                    max_value = np.inf
+                button_space["delta_" + str(len(delta_buttons))] = gym.spaces.Box(
+                    -max_value,
+                    max_value,
+                    (1,),
+                    dtype=np.float32,
+                )
+                delta_buttons.append(button)
+            else:
+                binary_buttons.append(button)
+        # fix delta buttons to be first in the one hot encoded action
+        self.game.set_available_buttons(delta_buttons + binary_buttons)
+        self.delta_buttons = len(delta_buttons)
+        self.binary_buttons = len(binary_buttons)
+
+        if self.delta_buttons == self.binary_buttons == 0:
```

Comment:
Why this? I do not see a scenario where attack alone would be an useful button to press.

Reply:
Yeah you're right, it's not useful, should I throw an error or not even check ?

Reply:
Lets throw an error if there are no actions :)